src/Components/Server/src/Circuits/RemoteJSRuntime.cs

```diff
@@ -78,17 +79,32 @@ protected override void BeginInvokeJS(long asyncHandle, string identifier, strin
         {
             if (_clientProxy is null)
             {
-                throw new InvalidOperationException(
-                    "JavaScript interop calls cannot be issued at this time. This is because the component is being " +
-                    $"statically rendered. When prerendering is enabled, JavaScript interop calls can only be performed " +
-                    $"during the OnAfterRenderAsync lifecycle method.");
+                if (_permanentlyDisconnected)
+                {
+                    throw new JSDisconnectedException(
+                           "JavaScript interop calls cannot be issued at this time. This is because the circuit has disconnected " +
+                           "and is being disposed.");
```

Comment:
```suggestion
                   "JavaScript interop calls cannot be issued at this time. This is because the circuit has disconnected " +
                   "and is being disposed.");
```